packages/m2m-gateway/src/api/attributeApiConverter.ts

```diff
@@ -47,3 +47,73 @@ export function toM2MGatewayApiCertifiedAttribute({
     }
   }
 }
+
+export function toM2MGatewayApiDeclaredAttribute({
+  attribute,
+  logger,
+  mapThrownErrorsToNotFound = false,
+}: {
+  attribute: attributeRegistryApi.Attribute;
+  logger: Logger;
+  mapThrownErrorsToNotFound?: boolean;
+}): m2mGatewayApi.DeclaredAttribute {
+  try {
+    assertAttributeKindIs(
+      attribute,
+      attributeRegistryApi.AttributeKind.Values.DECLARED
+    );
+
+    return {
+      id: attribute.id,
+      description: attribute.description,
+      name: attribute.name,
+      createdAt: attribute.creationTime,
+    };
+  } catch (error) {
+    if (mapThrownErrorsToNotFound) {
+      logger.warn(
+        `Root cause for "Attribute not found" error: unexpected error while converting attribute: ${
+          error instanceof ApiError ? error.detail : error
+        }`
+      );
+      throw attributeNotFound(attribute);
+    } else {
+      throw error;
+    }
+  }
+}
+
+export function toM2MGatewayApiVerifiedAttribute({
+  attribute,
+  logger,
+  mapThrownErrorsToNotFound = false,
+}: {
+  attribute: attributeRegistryApi.Attribute;
+  logger: Logger;
+  mapThrownErrorsToNotFound?: boolean;
+}): m2mGatewayApi.VerifiedAttribute {
+  try {
+    assertAttributeKindIs(
+      attribute,
+      attributeRegistryApi.AttributeKind.Values.VERIFIED
+    );
+
+    return {
+      id: attribute.id,
+      description: attribute.description,
+      name: attribute.name,
+      createdAt: attribute.creationTime,
+    };
+  } catch (error) {
+    if (mapThrownErrorsToNotFound) {
+      logger.warn(
+        `Root cause for "Attribute not found" error: unexpected error while converting attribute: ${
+          error instanceof ApiError ? error.detail : error
+        }`
+      );
+      throw attributeNotFound(attribute);
+    } else {
+      throw error;
+    }
+  }
```

Comment:
This piece of code is now the same in three different functions in this file, consider extracting a dedicated function for it